### PR TITLE
updated security group target APIs

### PR DESCRIPTION
**Pull request by deepaksibm**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccXXX'

...
```

### CODE REVIEW

**Comment by deepaksibm** [comment]
Screen Shot 2021-07-16 at 7 16 23 PM
Screen Shot 2021-07-16 at 7 42 07 PM
Screen Shot 2021-07-16 at 8 01 08 PM
Screen Shot 2021-07-16 at 9 02 18 PM

File: ibm/resource_ibm_is_security_group_network_interface_attachment.go [hunk 1]
```diff
@@ -183,17 +184,27 @@ func resourceIBMISSecurityGroupNetworkInterfaceAttachmentRead(d *schema.Resource
...
 	nicID := parts[1]
 
-	getSecurityGroupNetworkInterfaceOptions := &vpcv1.GetSecurityGroupNetworkInterfaceOptions{
+	getSecurityGroupNetworkInterfaceOptions := &vpcv1.GetSecurityGroupTargetOptions{
 		SecurityGroupID: &sgID,
 		ID:              &nicID,
 	}
-	instanceNic, response, err := sess.GetSecurityGroupNetworkInterface(getSecurityGroupNetworkInterfaceOptions)
+	secGroupTarget, response, err := sess.GetSecurityGroupTarget(getSecurityGroupNetworkInterfaceOptions)
 	if err != nil {
```
Comment (hkantare) [inline ibm/resource_ibm_is_security_group_network_interface_attachment.go:192]: add a condition if err != nil || secGroupTarget != nil  else if secGroupTarget is nil then will fail
instance_id := strings.Split(*secGroupTarget.(*vpcv1.SecurityGroupTargetReference).Href, "/")[5]

File: website/docs/r/is_security_group_network_interface_attachment.html.markdown [hunk 2]
```diff
@@ -18,6 +18,7 @@ resource "ibm_is_security_group_network_interface_attachment" "sgnic" {
   network_interface = "6d6128aa-badc-45c4-bb0e-7c2c1c47be55"
 }
 ```
+**Note** This resource is deprecated. Use `ibm_is_security_group_target` to attach a network interface to a security group
```
Comment (astha-jain) [inline website/docs/r/is_security_group_network_interface_attachment.html.markdown:21]: @hkantare Should we add a date to the deprecation notice as a reminder to remove these items in 6 months time frame?

Comment (hkantare) [reply website/docs/r/is_security_group_network_interface_attachment.html.markdown:21]: https://www.terraform.io/docs/extend/best-practices/deprecations.html#provider-data-source-or-resource-removal

Please follow this procedure for deprecation and removal

File: ibm/provider_test.go [hunk 3]
```diff
@@ -344,7 +344,7 @@ func init() {
 	isImage = os.Getenv("IS_IMAGE")
 	if isImage == "" {
 		//isImage = "fc538f61-7dd6-4408-978c-c6b85b69fe76" // for classic infrastructure
-		isImage = "r006-5b05b4fe-bcbc-4309-ad45-3354813227a0" // for next gen infrastructure
+		isImage = "r134-9573cb89-984e-45a4-bfc5-df611bbb8e7d" // for next gen infrastructure
 		fmt.Println("[INFO] Set the environment variable IS_IMAGE for testing ibm_is_instance, ibm_is_floating_ip else it is set to default value 'r006-ed3f775f-ad7e-4e37-ae62-7199b4988b00'")
 	}
```
Comment (SunithaGudisagarIBM) [inline ibm/provider_test.go:350]: This can be removed

**Comment by SunithaGudisagarIBM** [comment]
@hkantare I have reviewed this PR, From my end this looks fine to me

**Pull request merged**